src/main/java/seedu/address/model/ResumeBook.java

```diff
@@ -225,7 +226,14 @@ public void deleteInternship(Internship key) {
         internships.remove(key);
         for (Item item : resumes) {
             Resume resume = (Resume) item;
-            resume.getInternshipIds().remove(Integer.valueOf(id));
+            Resume newresume = new Resume(resume.getName(), resume.getId(), resume.getTags());
+            List<Integer> newInternshipList = new ArrayList<>();
+            newInternshipList.addAll(resume.getInternshipIds());
+            newresume.setInternshipIds(newInternshipList);
+            newresume.setProjectIds(resume.getProjectIds());
+            newresume.setSkillIds(resume.getSkillIds());
+            newresume.getInternshipIds().remove(Integer.valueOf(id));
```

Comment:
Adding white space for better readability
```suggestion

            newResume.setInternshipIds(newInternshipList);
            newResume.setProjectIds(resume.getProjectIds());
            newResume.setSkillIds(resume.getSkillIds());
            newResume.getInternshipIds().remove(Integer.valueOf(id));
            
```